### ISSUE TITLE
On first launch default orientation isn't honored

### ISSUE THREAD

**Issue by teckel12** [teckel12/LuaTelemetry#314]
Flying Horus/Crossfire today confirmed this.  I couldn't even change the view with the T6 button.  On second arming, no problem at all.  Very strange.  I didn't even do a flight reset, so it's not the reset function.  Something with the arm/disarm the first time doesn't work, second time works just fine.  Strange...

**Comment by teckel12** [comment]
Tried this extensively today, it only happened on Crossfire (and every time).  Trim 6 also never worked.  With SBUS, it worked every time, and trim 6 worked perfectly too.  Very odd.

Tim

**Comment by teckel12** [comment]
I believe this is fixed. The bug was a difficult to track down, but by refactoring the logic it sorted itself out.

**Comment by teckel12** [comment]
Nope!  That didn't fix it!  So strange, it worked perfectly every time for a day, then the next it failed every time with no changes.  More sifting through code 😢

**Comment by teckel12** [comment]
Figured it out! And it makes sense why sometimes it would always work, and other times it would never work.  It had to do with the orientation of the quad at launch.  I typically launch from a few places always pointing in the same direction (west and south).  But, when launching from the north, it would work.  It had to do with the Crossfire YAW value for orientation.  It's sometimes negative, which isn't a big deal as the algebra doesn't care.  But, I was using a `-1` setting as a flag for launch orientation not being set.  Instead of looking for just `-1` I simply looked for launch orientation less than zero (because on S.Port orientation is always 0-359).  But on Crossfire it was negative which threw off my check for home orientation not being set.

Anyway, no one probably even understands what I'm rambling about.  This was just a therapeutic reply as it was driving me mad.
